### ISSUE TITLE
recent addition of aws sqs instrumentation causing runtime failures when enabled

### ISSUE THREAD

**Issue by peterabbott** [elastic/apm-agent-nodejs#3746]
**Describe the bug**

**To Reproduce**

Using 4.1.0 of the agent. When publishing usign the SQS client and APM enabled, without any modifications. 

Getting error following error. Not been able to get nested error out yet. 

```
Error: Cannot convert undefined or null to object
```

The only different between the versions 4.0.0 and 4.1.0 is the introduction of aws-sqs instrumentation. Will post more details if we get anything out of the nested error. For now we can disable aws instrumentations to avoid renovate doignan auto bump again and breaking things.

**Expected behavior**

**Environment (please complete the following information)**

- OS: Lambda, ECS
- Node.js version: 18
- APM Server version: not applicable
- Agent version: : 4.1.0

**How are you starting the agent? (please tick one of the boxes)**

- [x] Calling `agent.start()` directly (e.g. `require('elastic-apm-node').start(...)`)
- [ ] Requiring `elastic-apm-node/start` from within the source code
- [ ] Starting node with `-r elastic-apm-node/start`

**Additional context**

Agent config options: 
<details>
  <summary>Click to expand</summary>

```
{
  ...baseConfig,
  // leaving these in to support using with lambda-executor for local testing
  serviceName: appName,
  active: process.env.NODE_ENV === 'production',
  disableInstrumentations: [],
  environment: environment + '-' + zone,
  globalLabels: {
    deployEnvironment: environment,
    zoneName: zone,
    serviceName: appName
  }
}
```
</details>

`package.json` dependencies:
<details>
  <summary>Click to expand</summary>

```
    "elastic-apm-node": "4.1.0"
```
</details>

**Comment by peterabbott** [comment]
Guess it was not picked up in testing because of this one 
https://github.com/elastic/apm-agent-nodejs/issues/3719

**Comment by peterabbott** [comment]
Managed to recreate locally with out test harness. It is this line [here](https://github.com/elastic/apm-agent-nodejs/blob/98f4694ed9c122e8ef2b4f2587ffc7aeb91be9aa/lib/instrumentation/modules/%40aws-sdk/client-sqs.js#L72)
does not take into account that that message attributes might be undefined.

```
 error: TypeError: Cannot convert undefined or null to object
      at Function.keys (<anonymous>)
      at  /XXXXXXXXXXX/elastic-apm-node/lib/instrumentation/modules/@aws-sdk/client-sqs.js:72:39
````

**Comment by david-luna** [comment]
Hi @peterabbott 

Thanks for reporting the issue and provide a diagnose :)

The instrumentation is checking for `MessageAttributes` being defined on `SendMessage` command
```js
if (commandName === 'SendMessage' && input.MessageAttributes)
```

For `SendMessageBatch` command is checking that `Entries` property is an array but not that the items inside are defined so I think this is where `undefined`s are coming from.

I'll start working with the fix today

**Comment by david-luna** [comment]
@peterabbott fix is available in latest release (v4.2.0). Thanks again for filing the issue :)
